src/main/java/org/jabref/gui/groups/GroupTreeViewModel.java

```diff
@@ -281,20 +281,36 @@ public void removeGroupKeepSubgroups(GroupNodeViewModel group) {
      * Removes the specified group and its subgroups (after asking for confirmation).
      */
     public void removeGroupAndSubgroups(GroupNodeViewModel group) {
-        boolean confirmed = dialogService.showConfirmationDialogAndWait(
-                Localization.lang("Remove group and subgroups"),
-                Localization.lang("Remove group \"%0\" and its subgroups?", group.getDisplayName()),
-                Localization.lang("Remove"));
+        boolean confirmed;
+        if (selectedGroups.size() <= 1) {
+            confirmed = dialogService.showConfirmationDialogAndWait(
+                    Localization.lang("Remove group and subgroups"),
+                    Localization.lang("Remove group \"%0\" and its subgroups?", group.getDisplayName()),
+                    Localization.lang("Remove"));
+        } else {
+            confirmed = dialogService.showConfirmationDialogAndWait(
+                    Localization.lang("Remove groups and subgroups"),
+                    Localization.lang("Remove all selected groups and their subgroups?"),
+                    Localization.lang("Remove all"));
+        }
+
         if (confirmed) {
             // TODO: Add undo
             // final UndoableAddOrRemoveGroup undo = new UndoableAddOrRemoveGroup(groupsRoot, node, UndoableAddOrRemoveGroup.REMOVE_NODE_AND_CHILDREN);
             // panel.getUndoManager().addEdit(undo);
 
-            removeGroupsAndSubGroupsFromEntries(group);
-
-            group.getGroupNode().removeFromParent();
+            GroupNodeViewModel[] selectedGroupNodes = new GroupNodeViewModel[selectedGroups.size()];
+            selectedGroupNodes = selectedGroups.toArray(selectedGroupNodes);
+            for (GroupNodeViewModel eachNode : selectedGroupNodes) {
+                removeGroupsAndSubGroupsFromEntries(eachNode);
+                eachNode.getGroupNode().removeFromParent();
+            }
```

Comment:
why are you creating an array first? You should be able to iterate through the ArrayList selectedGroups directly...
You can probably work with a new ArrayList<>(selectedGroups) too...

Reply:
Thank you for your feedback @calixtus.
The first solution came into my mind was C type array to prevent skip element after deletion. Definitely a rookie mistake from me.
I have updated with the use of ArrayList<> instead of array[].
Cheers.